game_board/database/test_db.py

```diff
@@ -0,0 +1,163 @@
+"""
+
+from game_board.database import game_board_db as mongo
+from django.test import TestCase
+
+class BColors:
+    # Colors for printing
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKCYAN = '\033[96m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+board = {
+    "game_id": "60afce36-085a-11eb-b6ab-acde48001122",
+    "graph": {
+        "nodes": "node4(node2(node3)(node1))(node6(node5))",
+        "node_points": {
+            "node1": 1,
+            "node2": 2,
+            "node3": 3,
+            "node4": 4,
+            "node5": 5,
+            "node6": 6
+        },
+        "gold_node": "node5",
+        "balanced": True
+    },
+    "player_ids": [
+        "id2",
+        "id3",
+        "id4",
+        "id5"
+    ],
+    "player_names": [
+        "naomi",
+        "kulsoom",
+        "nick",
+        "ryan"
+    ],
+    "player_points": {
+        "id2": 2,
+        "id3": 2,
+        "id4": 3,
+        "id5": 10
+    },
+    "turn": "id2",
+    "cards": {
+        "id2": [
+            "card1",
+            "card2",
+            "card3"
+        ],
+        "id3": [
+            "card1",
+            "card2",
+            "card3"
+        ],
+        "id4": [
+            "card1",
+            "card2",
+            "card3"
+        ],
+        "id5": [
+            "card1",
+            "card2",
+            "card3"
+        ]
+    },
+    "gold_node": False,
+    "difficulty": "Medium",
+    "num_players": 4,
+    "online": True,
+    "curr_data_structure": "AVL",
+    "selected_data_structures": [
+        "AVL",
+        "Stack"
+    ],
+    "timed_game": False,
+    "seconds_until_next_ds": 60,
+    "time_created": "07/10/2020 00:05:47",
+    "end_game": False
+}
+
+FAIL_PHRASE = 'nah bro idk about it'
+
+class DBCreate(TestCase):
+    def test_create(self):
+        """ The gamboard document was created in the database """
+        #Make sure the game is cleared out
+        mongo.remove_game(board["game_id"])
+        created_game = mongo.create_game( board )
+
+        self.assertEqual( created_game, "60afce36-085a-11eb-b6ab-acde48001122", msg=f'{BColors.FAIL}\t[-]\tGame was not created in the Database!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass gameboard database save.{BColors.ENDC}")
+
+    def test_duplicatecreate(self):
+        """ The document is a duplicate and should fail creation in the database """
+        created_game = mongo.create_game( board )
+
+        self.assertEqual( created_game, FAIL_PHRASE, msg=f'{BColors.FAIL}\t[-]\tGame was not created in the Database!{BColors.ENDC}')
+        print(f"{BColors.OKGREEN}\t[+]\tPass gameboard database save duplicate.{BColors.ENDC}")
+
+class DBRead(TestCase):
+    def test_read_correct(self):
+        """ The document was read from the database """
+        read_game = mongo.read_game( "60afce36-085a-11eb-b6ab-acde48001122" )
```

Comment:
This test might fail if 60afce36-085a-11eb-b6ab-acde48001122 does not exist. I recommend adding ```setUp()``` and ```tearDown()``` to this class which creates a game temporarily. A similar approach for other classes could be good as well. [See my code here for an example ](https://github.com/MaksimEkin/data-structures-game/blob/develop/game_board/api/tests_api.py#L167).

Reply:
Resolved in commit 3822b7384a745e576c538c02b6de0ede420990b9

Reply:
Resolved in commit 3822b7384a745e576c538c02b6de0ede420990b9